bin/antigen.zsh

```diff
@@ -805,27 +805,57 @@ antigen-theme () {
         add-zsh-hook -d "${hook}" "vcs_info"  # common in omz themes
     done
 }
+# Updates the bundles or a single bundle.
+#
+# Usage
+#    antigen-update [example/bundle]
+#
+# Returns
+#    Nothing. Performs a `git pull`.
 antigen-update () {
+    local bundle="$1"
+
     # Update your bundles, i.e., `git pull` in all the plugin repos.
     date >! $ADOTDIR/revert-info
 
     # Clear log
     :> $_ANTIGEN_LOG_PATH
 
-    -antigen-echo-record |
-        awk '$4 == "true" {print $1}' |
-        sort -u |
-        while read url; do
-            local clone_dir="$(-antigen-get-clone-dir "$url")"
-            if [[ -d "$clone_dir" ]]; then
-                (echo -n "$clone_dir:"
-                    cd "$clone_dir"
-                    git rev-parse HEAD) >> $ADOTDIR/revert-info
-            fi
+    if [[ ! $bundle == "" ]]; then
+        bundle_url=$(-antigen-resolve-bundle-url "$bundle")
+        if [[ ${(MS)_ANTIGEN_BUNDLE_RECORD##$bundle_url} ]]; then
+            -antigen-update-bundle "$bundle_url"
+        else
+            echo "Bundle not found in record. Try 'antigen bundle $bundle' first."
+            return 1
+        fi
+    else
+        -antigen-echo-record |
+            awk '$4 == "true" {print $1}' |
+            sort -u | while read url; do
+                -antigen-update-bundle $url
+            done
+    fi
+}
 
-            # update=true verbose=true
-            -antigen-ensure-repo "$url" true true
-        done
+# Updates a bundle performing a `git pull`.
+#
+# Usage
+#    -antigen-update-bundle https://github.com/example/bundle.zsh
+#
+# Returns
+#    Nothing. Performs a `git pull`.
+-antigen-update-bundle () {
+    local url="$1"
```

Comment:
Despite being a "private" function it should throw an error if "url" is not set, otherwise it will end up in `-antigen-ensure-repo` function.